common/numerated/src/tree.rs

```diff
@@ -405,10 +399,10 @@ impl<T: Numerated> IntervalsTree<T> {
     /// Number of points in tree set.
     ///
     /// Complexity: `O(n)`, where `n` is amount of intervals in `self`.
-    pub fn points_amount(&self) -> Option<T::N> {
-        let mut res = T::N::zero();
+    pub fn points_amount(&self) -> Option<T::Distance> {
```

Comment:
Shouldn't we consider an option to return usize here and for similar?

Reply:
its not the distance actually

Reply:
It's distance, because it's sum of distances.
`usize` - it's same as u32 or u64, and so, it's not suitable to count all points for corresponding u32 or u64 type.